frame/support/src/traits/hooks.rs

```diff
@@ -359,15 +359,22 @@ pub trait Hooks<BlockNumber> {
 	fn integrity_test() {}
 }
 
+/// Marker trait, indicating that something is the GenesisConfig for Pallet.
+pub trait PalletGenesisConfig<T, I = ()> {}
+
 /// A trait to define the build function of a genesis config, T and I are placeholder for pallet
 /// trait and pallet instance.
 pub trait GenesisBuild<T, I = ()>: Default + sp_runtime::traits::MaybeSerializeDeserialize {
 	/// The build function is called within an externalities allowing storage APIs.
 	/// Thus one can write to storage using regular pallet storages.
 	fn build(&self);
+}
 
+#[cfg(feature = "std")]
+/// A trait to define functions to build the  storage for a genesis config, T and I are placeholder
+/// for pallet trait and pallet instance.
+pub trait GenesisBuildStorage<T, I = ()>: GenesisBuild<T, I> {
```

Comment:
I don't really understand, you but in another trait these method so you can implement `GenesisBuild` on `RuntimeGenesisConfig` ?

I'm not sure it worth it, why not create a new trait for `RuntimeGenesisConfig` specifically, `GenesisBuild` was meant for pallets and thus the signature is quite specific with `<T, I>`, I think it is better to keep traits and implementation of `RuntimeGenesisConfig` apart, no ?

Reply:
Sorry for my bad suggestion, I agree what other said in the other PR, maybe this PR improve on the situation with a unique proper GenesisBuild, we could eventually get rid of the generics once with have decl_storage removed (is it already ?) and we can keep the other method for not breaking peoples tests in another trait or another way